File: java/src/binary_search/LC_278_FirstBadVersion.java
package binary_search;
// https://leetcode.com/problems/first-bad-version

public class LC_278_FirstBadVersion {

    boolean isBadVersion(int version) {
        return true;
    }

    public int firstBadVersion(int n) {
        int left = 0, right = n;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (isBadVersion(mid))
                right = mid;
            else
                left = mid + 1;
        }
        return left;
    }
}

File: python/binary_search/LC_278_FirstBadVersion.py
# https: // leetcode.com/problems/first-bad-version

def isBadVersion(version: int) -> bool:
    pass


class Solution:
    def firstBadVersion(self, n: int) -> int:
        left, right = 0, n
        while left < right:
            mid = left + (right - left) // 2
            is_bad = isBadVersion(mid)
            if is_bad:
                right = mid
            else:
                left = mid + 1

        return left
